### DISCUSSION TITLE
When retry is closed, it always throws an error seems to ignore the configuration about throwOnError?

### DISCUSSION THREAD

**Discussion by StringKe** [TanStack/query#7489] [Q&A] [answered]
```ts
const { data, refetch, isSuccess, isError } = useQuery({
        queryKey: ['test'],
        queryFn: async () => {
            throw new Error('test');
        },
        retry: false,
        throwOnError: false,
});
```

Using it this way causes errors to be thrown and ErrorBoundary to accept them, leading to unexpected UI fallbacks.

**Comment by TkDodo** [comment]
Shouldn't be the case - please show a minimal reproduction

**Reply by StringKe** [reply]
By checking this issue seems to be caused by tanstack/route.

I've tried creating several demos without this problem and I'm checking my dependency versions.

**Reply by StringKe** [reply, marked as answer]
https://github.com/TanStack/router/issues/857#issuecomment-2136645861
